sklearn/pipeline.py

```diff
@@ -981,6 +985,11 @@ def _update_transformer_list(self, transformers):
                                      else next(transformers))
                                     for name, old in self.transformer_list]
 
+    @property
+    def n_features_in_(self):
```

Comment:
don't we want to raise an `AttributeError` before `fit` in these two cases as well?

Reply:
`AttributeError` is going to be raised by `transformer_list[0][1]` (you can check `test_n_features_in_pipeline()`).

One funky side effect is that if `transformer_list[0][1]` is fitted but the rest of the pipeline isn't fitted, the pipeline still has the attribute. I guess that's a case that we don't want to worry about, since it's typically never happening in reasonable scenarios? I documented it in the tests though.

Reply:
Right, good then.